@kiva/kv-components/vue/KvPagination.vue

```diff
@@ -0,0 +1,198 @@
+		},
+		offset: {
+			type: Number,
+			default: 0,
+			validator: (value) => value >= 0,
+		},
+		extraPages: {
+			type: Number,
+			default: 3,
+			validator: (value) => value > 0,
+		},
+		scrollToTop: {
+			type: Boolean,
+			default: true,
+		},
+		kvTrackFunction: {
+			type: Function,
+			default: () => {},
+		},
+		trackEventCategory: {
+			type: String,
+			default: '',
+		},
+	},
+	data() {
+		return {
+			mdiChevronLeft,
+			mdiChevronRight,
+		};
+	},
+	computed: {
+		current() {
+			const page = this.offset / this.limit;
+
+			// This component uses a 0-based page index
+			return page < this.totalPages ? page : 0;
+		},
+		totalPages() {
+			return Math.ceil(this.total / this.limit);
+		},
+		numbers() {
+			// If less than the max, there will be no ellipsis, so just return the numbers
+			if (this.totalPages < (this.extraPages + 2)) {
+				return this.range(0, this.totalPages - 1);
+			}
+
+			const numbers = [];
+
+			// Add the 'middle' block of numbers based upon the current page
+			if ([0, 1, 2].includes(this.current)) {
+				numbers.push(...this.range(1, this.extraPages));
+			} else if ([this.totalPages - 3, this.totalPages - 2, this.totalPages - 1]
+				.includes(this.current)) {
+				numbers.push(...this.range(this.totalPages - this.extraPages - 1, this.totalPages - 2));
+			} else {
+				const delta = Math.floor(this.extraPages / 2);
+				numbers.push(...this.range(this.current - delta, this.current + delta));
+			}
+
+			// Add a placeholder for first ellipsis
+			if (numbers[1] !== 2) {
+				numbers.splice(0, 0, -1);
+			}
+
+			// Add a placeholder for second ellipsis
+			const totalNumbers = numbers.length;
+			if (numbers[totalNumbers - 1] !== this.totalPages - 2) {
+				numbers.splice(totalNumbers, 0, -1);
+			}
+
+			// Add first and last pages
+			numbers.unshift(0);
+			numbers.push(this.totalPages - 1);
+
+			return numbers;
+		},
+	},
+	methods: {
+		range(start, end) {
+			return [...Array(end - start + 1)].map((_, n) => n + start);
+		},
+		isCurrent(number) {
+			return number === this.current;
+		},
+		isEllipsis(number) {
+			return number === -1;
+		},
+		linkClass(number) {
+			return { 'tw-text-tertiary': this.isCurrent(number), 'tw-pointer-events-none': this.isCurrent(number) };
+		},
+		pageChange(number) {
+			if (this.scrollToTop && window.scrollTo) {
+				window.scrollTo({ top: 0, left: 0, behavior: 'smooth' });
+			}
+
+			this.$emit('page-changed', { pageOffset: number * this.limit });
+		},
+		clickPage(number) {
+			this.pageChange(number);
+
+			this.kvTrackFunction(this.trackEventCategory, 'click-page-pager', number + 1);
+		},
+		clickPrevious() {
+			const previous = this.current - 1;
+
+			this.pageChange(previous);
+
+			this.kvTrackFunction(this.trackEventCategory, 'click-previous-pager', previous + 1);
+		},
+		clickNext() {
+			const next = this.current + 1;
+
+			this.pageChange(next);
+
+			this.kvTrackFunction(this.trackEventCategory, 'click-next-pager', next + 1);
```

Comment:
For previous and next please use "click" for the action followed by "pagination-previous" or "pagination-next" and the value after that.